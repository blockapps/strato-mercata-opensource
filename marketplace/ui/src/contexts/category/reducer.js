import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.createCategory:
      return {
        ...state,
        isCreateCategorySubmitting: true,
      };
    case actionDescriptors.createCategorySuccessful:
      return {
        ...state,
        category: action.payload,
        isCreateCategorySubmitting: false,
      };
    case actionDescriptors.createCategoryFailed:
      return {
        ...state,
        error: action.error,
        isCreateCategorySubmitting: false,
      };
    case actionDescriptors.fetchCategory:
      return {
        ...state,
        iscategorysLoading: true,
      };
    case actionDescriptors.fetchCategorySuccessful:
      return {
        ...state,
        categorys: action.payload,
        iscategorysLoading: false,
      };
    case actionDescriptors.fetchCategoryFailed:
      return {
        ...state,
        error: action.error,
        iscategorysLoading: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
