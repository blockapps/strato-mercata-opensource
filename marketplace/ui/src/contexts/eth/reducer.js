import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.fetchETHSTAddress:
      return {
        ...state,
      };
    case actionDescriptors.fetchETHSTAddressSuccessful:
      return {
        ...state,
        ethstAddress: action.payload,
      };
    case actionDescriptors.fetchETHSTAddressFailed:
      return {
        ...state,
      };
    case actionDescriptors.addHash:
      return {
        ...state,
        isAddingHash: true,
      };
    case actionDescriptors.addHashSuccessful:
      return {
        ...state,
        isAddingHash: false,
      };
    case actionDescriptors.addHashFailed:
      return {
        ...state,
        isAddingHash: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
