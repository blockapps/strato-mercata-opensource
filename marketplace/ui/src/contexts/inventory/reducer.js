import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.createInventory:
      return {
        ...state,
        isCreateInventorySubmitting: true,
      };
    case actionDescriptors.createInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.createInventoryFailed:
      return {
        ...state,
        error: action.error,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.fetchInventory:
      return {
        ...state,
        isInventoriesLoading: true,
      };
    case actionDescriptors.fetchInventorySuccessful:
      return {
        ...state,
        inventories: action.payload.data,
        inventoriesTotal: action.payload.count,
        isInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventoryFailed:
      return {
        ...state,
        error: action.error,
        isInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventoryForUser:
      return {
        ...state,
        isUserInventoriesLoading: true,
      };
    case actionDescriptors.fetchInventoryForUserSuccessful:
      return {
        ...state,
        userInventories: action.payload.data,
        userInventoriesTotal: action.payload.count,
        isUserInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventoryForUserFailed:
      return {
        ...state,
        error: action.error,
        isUserInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventorySearch:
      return {
        ...state,
        isInventoriesLoading: true,
      };
    case actionDescriptors.fetchInventorySearchSuccessful:
      return {
        ...state,
        inventories: action.payload.data,
        inventoriesTotal: action.payload.count,
        isInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventorySearchFailed:
      return {
        ...state,
        error: action.error,
        isInventoriesLoading: false,
      };
    case actionDescriptors.updateInventory:
      return {
        ...state,
        isinventoryUpdating: true,
      };
    case actionDescriptors.updateInventorySuccessful:
      return {
        ...state,
        inventoryUpdateObject: action.payload,
        isinventoryUpdating: false,
      };
    case actionDescriptors.updateInventoryFailed:
      return {
        ...state,
        error: action.error,
        isinventoryUpdating: false,
      };
    case actionDescriptors.updateSale:
      return {
        ...state,
        issaleUpdating: true,
      };
    case actionDescriptors.updateSaleSuccessful:
      return {
        ...state,
        saleUpdateObject: action.payload,
        issaleUpdating: false,
      };
    case actionDescriptors.updateSaleFailed:
      return {
        ...state,
        error: action.error,
        issaleUpdating: false,
      };
    case actionDescriptors.listInventory:
      return {
        ...state,
        isListing: true,
      };
    case actionDescriptors.listInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isListing: false,
      };
    case actionDescriptors.listInventoryFailed:
      return {
        ...state,
        error: action.error,
        isListing: false,
      };
    case actionDescriptors.unlistInventory:
      return {
        ...state,
        isUnlisting: true,
      };
    case actionDescriptors.unlistInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isUnlisting: false,
      };
    case actionDescriptors.unlistInventoryFailed:
      return {
        ...state,
        error: action.error,
        isUnlisting: false,
      };

     // ------------------------------------------------------------------------------------------------------
    case actionDescriptors.stakeInventory:
      return {
        ...state,
        isStaking: true
      };
    case actionDescriptors.stakeInventorySuccessful:
      return {
        ...state,
        isStaking: false
      };
    case actionDescriptors.stakeInventoryFailed:
      return {
        ...state,
        error: action.error,
        isStaking: false
      };
    
    case actionDescriptors.unstakeInventory:
      return {
        ...state,
        isUnstaking: true
      };
    case actionDescriptors.unstakeInventorySuccessful:
      return {
        ...state,
        isUnstaking: false
      };
    case actionDescriptors.unstakeInventoryFailed:
      return {
        ...state,
        error: action.error,
        isUnstaking: false
      };
    
    case actionDescriptors.borrow:
      return {
        ...state,
        isBorrowing: true
      };
    case actionDescriptors.borrowSuccessful:
      return {
        ...state,
        isBorrowing: false
      };
    case actionDescriptors.borrowFailed:
      return {
        ...state,
        error: action.error,
        isBorrowing: false
      };
    
    case actionDescriptors.repay:
      return {
        ...state,
        isRepaying: true
      };
    case actionDescriptors.repaySuccessful:
      return {
        ...state,
        isRepaying: false
      };
    case actionDescriptors.repayFailed:
        return {
          ...state,
          error: action.error,
          isRepaying: false
        };

    case actionDescriptors.getReserve:
        return {
          ...state,
          isReserveLoading: true
        };
    case actionDescriptors.getReserveSuccessful:
        return {
          ...state,
          reserve: action.payload,
          isReserveLoading: false
        };
    case actionDescriptors.getReserveFailed:
        return {
          ...state,
          error: action.error,
          isReserveLoading: false
        };
    
    case actionDescriptors.getAllReserve:
        return {
          ...state,
          isReservesLoading: true
        };
    case actionDescriptors.getAllReserveSuccessful:
        return {
          ...state,
          reserves: action.payload,
          isReservesLoading: false
        };
    case actionDescriptors.getAllReserveFailed:
        return {
          ...state,
          error: action.error,
          isReservesLoading: false
        };
    
    case actionDescriptors.getEscrowForAsset:
        return {
          ...state,
          isEscrowLoading: true
        };
    case actionDescriptors.getEscrowForAssetSuccessful:
        return {
          ...state,
          escrow: action.payload,
          isEscrowLoading: false
        };
    case actionDescriptors.getEscrowForAssetFailed:
        return {
          ...state,
          error: action.error,
          isEscrowLoading: false
        };

    case actionDescriptors.getUserCataRewards:
        return {
          ...state,
          isUserCataRewardsLoading: true
        };
    case actionDescriptors.getUserCataRewardsSuccessful:
        return {
          ...state,
          totalCataReward: action.payload.totalCataReward,
          dailyCataReward: action.payload.dailyCataReward,
          isUserCataRewardsLoading: false
        };
    case actionDescriptors.getUserCataRewardsFailed:
        return {
          ...state,
          error: action.error,
          isUserCataRewardsLoading: false
        };
    
    case actionDescriptors.getOracle:
          return {
            ...state,
            isFetchingOracle: true
          };
    case actionDescriptors.getOracleSuccessful:
          return {
            ...state,
            oracle: action.payload,
            isFetchingOracle: false
          };
    case actionDescriptors.getOracleFailed:
          return {
            ...state,
            error: action.error,
            isFetchingOracle: false
          };    
    // ------------------------------------------------------------------------------------------------------
    case actionDescriptors.resellInventory:
      return {
        ...state,
        isReselling: true,
      };
    case actionDescriptors.resellInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isReselling: false,
      };
    case actionDescriptors.resellInventoryFailed:
      return {
        ...state,
        error: action.error,
        isReselling: false,
      };
    case actionDescriptors.fetchSupportedTokens:
      return {
        ...state,
        isFetchingTokens: true,
      };
    case actionDescriptors.fetchSupportedTokensSuccessful:
      return {
        ...state,
        supportedTokens: action.payload,
        isFetchingTokens: false,
      };
    case actionDescriptors.fetchSupportedTokensFailed:
      return {
        ...state,
        error: action.error,
        isFetchingTokens: false,
      };
    case actionDescriptors.bridgeInventory:
      return {
        ...state,
        isBridging: true,
      };
    case actionDescriptors.bridgeInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isBridging: false,
      };
    case actionDescriptors.bridgeInventoryFailed:
      return {
        ...state,
        error: action.error,
        isBridging: false,
      };
    case actionDescriptors.transferInventory:
      return {
        ...state,
        isTransferring: true,
      };
    case actionDescriptors.transferInventorySuccessful:
      return {
        ...state,
        isTransferring: false,
      };
    case actionDescriptors.transferInventoryFailed:
      return {
        ...state,
        error: action.error,
        isTransferring: false,
      };
    case actionDescriptors.fetchItemTransfers:
      return {
        ...state,
        isFetchingItemTransfers: true,
      };
    case actionDescriptors.fetchItemTransfersSuccessful:
      return {
        ...state,
        itemTransfers: action.payload.transfers,
        totalItemsTransfered: action.payload.total,
        isFetchingItemTransfers: false,
      };
    case actionDescriptors.fetchItemTransfersFailed:
      return {
        ...state,
        error: action.error,
        isFetchingItemTransfers: false,
      };
    case actionDescriptors.fetchInventoryOwnershipHistory:
      return {
        ...state,
        isInventoryOwnershipHistoryLoading: true,
      };
    case actionDescriptors.fetchInventoryOwnershipHistorySuccessful:
      return {
        ...state,
        inventoryOwnershipHistory: action.payload,
        isInventoryOwnershipHistoryLoading: false,
      };
    case actionDescriptors.fetchInventoryOwnershipHistoryFailed:
      return {
        ...state,
        error: action.error,
        isInventoryOwnershipHistoryLoading: false,
      };
    case actionDescriptors.fetchInventoryDetail:
      return {
        ...state,
        isInventoryDetailsLoading: true,
      };
    case actionDescriptors.fetchInventoryDetailSuccessful:
      return {
        ...state,
        inventoryDetails: action.payload,
        isInventoryDetailsLoading: false,
      };
    case actionDescriptors.fetchInventoryDetailFailed:
      return {
        ...state,
        error: action.error,
        isInventoryDetailsLoading: false,
      };
    case actionDescriptors.uploadImage:
      return {
        ...state,
        isUploadImageSubmitting: true,
      };
    case actionDescriptors.uploadImageSuccessful:
      return {
        ...state,
        uploadedImg: action.payload,
        isUploadImageSubmitting: false,
      };
    case actionDescriptors.uploadImageFailed:
      return {
        ...state,
        error: action.error,
        isUploadImageSubmitting: false,
      };
    case actionDescriptors.createItem:
      return {
        ...state,
        isCreateInventorySubmitting: true,
      };
    case actionDescriptors.createItemSuccessful:
      return {
        ...state,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.createItemFailed:
      return {
        ...state,
        error: action.error,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.fetchPriceHistory:
      return {
        ...state,
        isFetchingPriceHistory: true,
      };
    case actionDescriptors.fetchPriceHistorySuccessful:
      return {
        ...state,
        isFetchingPriceHistory: false,
        priceHistory: action.payload,
      };
    case actionDescriptors.fetchPriceHistoryFailed:
      return {
        ...state,
        isFetchingPriceHistory: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
