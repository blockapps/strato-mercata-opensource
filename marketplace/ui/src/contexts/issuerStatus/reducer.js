import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.requestReview:
      return {
        ...state,
        requestingReview: true,
      };
    case actionDescriptors.requestReviewSuccessful:
      return {
        ...state,
        requestingReview: false,
      };
    case actionDescriptors.requestReviewFailed:
      return {
        ...state,
        requestingReview: false,
      };
    case actionDescriptors.authorizeIssuer:
      return {
        ...state,
        changingIssuerStatus: true,
      };
    case actionDescriptors.authorizeIssuerSuccessful:
      return {
        ...state,
        changingIssuerStatus: false,
      };
    case actionDescriptors.authorizeIssuerFailed:
      return {
        ...state,
        changingIssuerStatus: false,
      };
    case actionDescriptors.deauthorizeIssuer:
      return {
        ...state,
        changingIssuerStatus: true,
      };
    case actionDescriptors.deauthorizeIssuerSuccessful:
      return {
        ...state,
        changingIssuerStatus: false,
      };
    case actionDescriptors.deauthorizeIssuerFailed:
      return {
        ...state,
        changingIssuerStatus: false,
      };
    case actionDescriptors.modifyAdmin:
      return {
        ...state,
        changingAdminStatus: true,
      };
    case actionDescriptors.modifyAdminSuccessful:
      return {
        ...state,
        changingAdminStatus: false,
      };
    case actionDescriptors.modifyAdminFailed:
      return {
        ...state,
        changingAdminStatus: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
