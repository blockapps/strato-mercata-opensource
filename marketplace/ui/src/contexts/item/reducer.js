import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.fetchItem:
      return {
        ...state,
        isItemsLoading: true,
      };
    case actionDescriptors.fetchItemSuccessful:
      return {
        ...state,
        items: action.payload,
        isItemsLoading: false,
      };
    case actionDescriptors.fetchItemFailed:
      return {
        ...state,
        error: action.error,
        isItemsLoading: false,
      };
    case actionDescriptors.fetchSerialNumbers:
      return {
        ...state,
        isSerialNumbersLoading: true,
      };
    case actionDescriptors.fetchSerialNumbersSuccessful:
      return {
        ...state,
        serialNumbers: action.payload,
        isSerialNumbersLoading: false,
      };
    case actionDescriptors.fetchSerialNumbersFailed:
      return {
        ...state,
        error: action.error,
        isSerialNumbersLoading: false,
      };
    case actionDescriptors.fetchItemOwnershipHistory:
      return {
        ...state,
        isOwnershipHistoryLoading: true,
      };
    case actionDescriptors.fetchItemOwnershipHistorySuccessful:
      return {
        ...state,
        ownershipHistory: action.payload,
        isOwnershipHistoryLoading: false,
      };
    case actionDescriptors.fetchItemOwnershipHistoryFailed:
      return {
        ...state,
        error: action.error,
        isOwnershipHistoryLoading: false,
      };
    case actionDescriptors.fetchItemTransfers:
      return {
        ...state,
        isFetchingItemTransfers: true,
      };
    case actionDescriptors.fetchItemTransfersSuccessful:
      return {
        ...state,
        itemTransfers: action.payload.transfers,
        totalItemsTransfered: action.payload.total,
        isFetchingItemTransfers: false,
      };
    case actionDescriptors.fetchItemTransfersFailed:
      return {
        ...state,
        error: action.error,
        isFetchingItemTransfers: false,
      };
    case actionDescriptors.fetchItemRawMaterials:
      return {
        ...state,
        isRawMaterialsLoading: true,
      };
    case actionDescriptors.fetchItemRawMaterialsSuccessful:
      return {
        ...state,
        rawMaterials: action.payload,
        isRawMaterialsLoading: false,
      };
    case actionDescriptors.fetchItemRawMaterialsFailed:
      return {
        ...state,
        error: action.error,
        isRawMaterialsLoading: false,
      };
    case actionDescriptors.setActualRawMaterials:
      return {
        ...state,
        actualRawMaterials: action.payload,
      };
    case actionDescriptors.transferOwnership:
      return {
        ...state,
        isTransferringItems: true,
      };
    case actionDescriptors.transferOwnershipSuccessful:
      return {
        ...state,
        isTransferringItems: false,
      };
    case actionDescriptors.transferOwnershipFailed:
      return {
        ...state,
        error: action.error,
        isTransferringItems: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
