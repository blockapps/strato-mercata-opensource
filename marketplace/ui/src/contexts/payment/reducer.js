import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.getPaymentServices:
      return {
        ...state,
        arePaymentServicesLoading: true,
      };
    case actionDescriptors.getPaymentServicesSuccessful:
      return {
        ...state,
        paymentServices: action.payload.data,
        paymentServicesTotal: action.payload.count,
        arePaymentServicesLoading: false,
      };
    case actionDescriptors.getPaymentServicesFailed:
      return {
        ...state,
        error: action.error,
        arePaymentServicesLoading: false,
      };
    case actionDescriptors.getNotOnboarded:
      return {
        ...state,
        areNotOnboardedLoading: true,
      };
    case actionDescriptors.getNotOnboardedSuccessful:
      return {
        ...state,
        notOnboarded: action.payload.data,
        notOnboardedTotal: action.payload.count,
        areNotOnboardedLoading: false,
      };
    case actionDescriptors.getNotOnboardedFailed:
      return {
        ...state,
        error: action.error,
        areNotOnboardedLoading: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
