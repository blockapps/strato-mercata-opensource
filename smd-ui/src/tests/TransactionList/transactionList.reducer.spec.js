import reducer from '../../components/TransactionList/transactionList.reducer';
import { transactions, updatedTransactions, last, error } from "./transactionListMock";
import {
  updateTx,
  preloadTx,
  UPDATE_TX,
  PRELOAD_TX,
  fetchTx,
  fetchTxSuccess,
  fetchTxFailure
} from '../../components/TransactionList/transactionList.actions';
import { deepClone } from '../helper/testHelper';

describe('TransactionList: reducer', () => {

  // INITIAL_STATE
  test('set initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  describe('transactions', () => {

    // UPDATE_TX
    test('update', () => {
      const action = preloadTx(transactions);
      const initialState = {
        tx: [],
        transactions: [],
        error: null,
      }
      expect(reducer(initialState, action)).toMatchSnapshot();
    })

    // PRELOAD_TX
    test('load', () => {
      const action = updateTx(updatedTransactions);
      const initialState = {
        tx: [],
        transactions: transactions,
        error: null,
      }
      expect(reducer(initialState, action)).toMatchSnapshot();
    })

    // FETCH_TX
    test('request', () => {
      const action = fetchTx(last);
      const initialState = {
        tx: [],
        transactions: [],
        error: null
      }
      expect(reducer(initialState, action)).toMatchSnapshot();
    })

    // FETCH_TX_SUCCESSFUL
    test('success', () => {
      const action = fetchTxSuccess(transactions);
      const initialState = {
        tx: [],
        transactions: [],
        error: null
      }
      expect(reducer(initialState, action)).toMatchSnapshot();
    })

    // FETCH_TX_FAILED
    test('failure', () => {
      const action = fetchTxFailure(error);
      const initialState = {
        tx: [],
        transactions: [],
        error: null
      }
      expect(reducer(initialState, action)).toMatchSnapshot();
    });
  });
});
