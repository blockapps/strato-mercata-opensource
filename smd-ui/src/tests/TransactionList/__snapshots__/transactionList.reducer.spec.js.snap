// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionList: reducer set initial state 1`] = `
Object {
  "error": null,
  "transactions": Array [],
  "tx": Array [],
}
`;

exports[`TransactionList: reducer transactions failure 1`] = `
Object {
  "error": "error",
  "transactions": Array [],
  "tx": Array [],
}
`;

exports[`TransactionList: reducer transactions load 1`] = `
Object {
  "error": null,
  "transactions": Array [
    Object {
      "blockNumber": 21,
      "codeOrData": "0e3668f8000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000026673000000000000000000000000000000000000000000000000000000000000",
      "from": "d027272d732f730f3fef139aa40293a3a5800fbe",
      "gasLimit": 100000000,
      "gasPrice": 1,
      "hash": "9c47c9d3284248a6f85b9192aea2214db97252c67850ebfebb0e33eb6237949b",
      "next": "",
      "nonce": 18,
      "origin": "API",
      "r": "1b720e3006b58e466ee7c6635bab4e627afab0c5ea26d1937ac378f8669ce781",
      "s": "6f8f8173644f7d6e847d0d95207b69f9e2834bde16e62a05f41f6bfd725ce0e4",
      "timestamp": "2017-12-14 12:42:13.017694 UTC",
      "to": "26acb720a5029f6e50b94e096e7df1d03340879f",
      "transactionType": "FunctionCall",
      "v": "1b",
      "value": "0",
    },
    Object {
      "blockNumber": 20,
      "codeOrData": "0e3668f8000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000026673000000000000000000000000000000000000000000000000000000000000",
      "from": "d027272d732f730f3fef139aa40293a3a5800fbe",
      "gasLimit": 100000000,
      "gasPrice": 1,
      "hash": "5e06d623d2ec0eaad0f8b5072a85c6bf5f042924d85d77b1ed566c502a4f4b82",
      "next": "",
      "nonce": 17,
      "origin": "API",
      "r": "68e0e1c9602a4166a94a230d1d6075df7b280eeb4a8ade51a9adbb68bf247ae9",
      "s": "f1caf64684d6f4ec2941c2583e78c004b6fb87665e521c46e407e446a976fd9",
      "timestamp": "2017-12-14 12:41:32.229929 UTC",
      "to": "26acb720a5029f6e50b94e096e7df1d03340879f",
      "transactionType": "FunctionCall",
      "v": "1c",
      "value": "0",
    },
    Object {
      "blockNumber": 19,
      "codeOrData": "6060604052346000575b6062806100176000396000f30060606040526000357c0100000000000000000000000000000000000000000000000000000000900463ffffffff1680632a1afcd914603c575b6000565b346000576046605c565b6040518082815260200191505060405180910390f35b6000548156",
      "from": "d027272d732f730f3fef139aa40293a3a5800fbe",
      "gasLimit": 100000000,
      "gasPrice": 1,
      "hash": "70018a76a7aa0e6d54565ae22264ac48773a52204c47fd0166b5a6df6e8f2a81",
      "next": "",
      "nonce": 16,
      "origin": "API",
      "r": "f5f15124e4d9395490989a6e6947492d40a3ba65afea0cce1584ca1bf0fde95d",
      "s": "38230fe3a6b0a629f3ea5f441b13ff784d9c52d0663476e3b066eb1b138fc248",
      "timestamp": "2017-12-13 07:47:05.998689 UTC",
      "transactionType": "Contract",
      "v": "1b",
      "value": "0",
    },
  ],
  "tx": Array [],
}
`;

exports[`TransactionList: reducer transactions request 1`] = `
Object {
  "error": null,
  "transactions": Array [],
  "tx": Array [],
}
`;

exports[`TransactionList: reducer transactions success 1`] = `
Object {
  "error": null,
  "transactions": Array [],
  "tx": Array [
    Object {
      "blockNumber": 21,
      "codeOrData": "0e3668f8000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000026673000000000000000000000000000000000000000000000000000000000000",
      "from": "d027272d732f730f3fef139aa40293a3a5800fbe",
      "gasLimit": 100000000,
      "gasPrice": 1,
      "hash": "9c47c9d3284248a6f85b9192aea2214db97252c67850ebfebb0e33eb6237949b",
      "next": "",
      "nonce": 18,
      "origin": "API",
      "r": "1b720e3006b58e466ee7c6635bab4e627afab0c5ea26d1937ac378f8669ce781",
      "s": "6f8f8173644f7d6e847d0d95207b69f9e2834bde16e62a05f41f6bfd725ce0e4",
      "timestamp": "2017-12-14 12:42:13.017694 UTC",
      "to": "26acb720a5029f6e50b94e096e7df1d03340879f",
      "transactionType": "FunctionCall",
      "v": "1b",
      "value": "0",
    },
    Object {
      "blockNumber": 20,
      "codeOrData": "0e3668f8000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000026673000000000000000000000000000000000000000000000000000000000000",
      "from": "d027272d732f730f3fef139aa40293a3a5800fbe",
      "gasLimit": 100000000,
      "gasPrice": 1,
      "hash": "5e06d623d2ec0eaad0f8b5072a85c6bf5f042924d85d77b1ed566c502a4f4b82",
      "next": "",
      "nonce": 17,
      "origin": "API",
      "r": "68e0e1c9602a4166a94a230d1d6075df7b280eeb4a8ade51a9adbb68bf247ae9",
      "s": "f1caf64684d6f4ec2941c2583e78c004b6fb87665e521c46e407e446a976fd9",
      "timestamp": "2017-12-14 12:41:32.229929 UTC",
      "to": "26acb720a5029f6e50b94e096e7df1d03340879f",
      "transactionType": "FunctionCall",
      "v": "1c",
      "value": "0",
    },
  ],
}
`;

exports[`TransactionList: reducer transactions update 1`] = `
Object {
  "error": null,
  "transactions": Array [
    Object {
      "blockNumber": 21,
      "codeOrData": "0e3668f8000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000026673000000000000000000000000000000000000000000000000000000000000",
      "from": "d027272d732f730f3fef139aa40293a3a5800fbe",
      "gasLimit": 100000000,
      "gasPrice": 1,
      "hash": "9c47c9d3284248a6f85b9192aea2214db97252c67850ebfebb0e33eb6237949b",
      "next": "",
      "nonce": 18,
      "origin": "API",
      "r": "1b720e3006b58e466ee7c6635bab4e627afab0c5ea26d1937ac378f8669ce781",
      "s": "6f8f8173644f7d6e847d0d95207b69f9e2834bde16e62a05f41f6bfd725ce0e4",
      "timestamp": "2017-12-14 12:42:13.017694 UTC",
      "to": "26acb720a5029f6e50b94e096e7df1d03340879f",
      "transactionType": "FunctionCall",
      "v": "1b",
      "value": "0",
    },
    Object {
      "blockNumber": 20,
      "codeOrData": "0e3668f8000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000026673000000000000000000000000000000000000000000000000000000000000",
      "from": "d027272d732f730f3fef139aa40293a3a5800fbe",
      "gasLimit": 100000000,
      "gasPrice": 1,
      "hash": "5e06d623d2ec0eaad0f8b5072a85c6bf5f042924d85d77b1ed566c502a4f4b82",
      "next": "",
      "nonce": 17,
      "origin": "API",
      "r": "68e0e1c9602a4166a94a230d1d6075df7b280eeb4a8ade51a9adbb68bf247ae9",
      "s": "f1caf64684d6f4ec2941c2583e78c004b6fb87665e521c46e407e446a976fd9",
      "timestamp": "2017-12-14 12:41:32.229929 UTC",
      "to": "26acb720a5029f6e50b94e096e7df1d03340879f",
      "transactionType": "FunctionCall",
      "v": "1c",
      "value": "0",
    },
  ],
  "tx": Array [],
}
`;
